pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -208,16 +213,66 @@ func (lb *ListenerBuilder) buildWaypointInboundConnectTerminate() *listener.List
 
 func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs []*model.Service) *listener.Listener {
 	ipMatcher := &matcher.IPMatcher{}
+	svcHostnameMap := &matcher.Matcher_MatcherTree_MatchMap{
+		Map: make(map[string]*matcher.Matcher_OnMatch),
+	}
 	chains := []*listener.FilterChain{}
 	portProtocols := map[int]protocol.Instance{}
+	getOrigDstSfs := func(ipAndPort string, isHttp bool) *listener.Filter {
+		celTemplate := `%%CEL('%s' in filter_state ? filter_state['%s']: r'%s')%%`
```

Comment:
is that so intelligent?

Reply:
Can you explain a little bit more?

Reply:
Ah we use fmt.sprintf later in go to populate the template with the ip address and filter state key